File: webflux-with-mongo-example/src/main/java/com/example/webfluxwithmongoexample/application/facade/user/UserFacade.java
package com.example.webfluxwithmongoexample.application.facade.user;

import com.example.webfluxwithmongoexample.domain.user.service.UserCommand;
import com.example.webfluxwithmongoexample.domain.user.service.UserInfo;
import com.example.webfluxwithmongoexample.domain.user.service.UserService;
import lombok.RequiredArgsConstructor;
import org.springframework.stereotype.Component;
import reactor.core.publisher.Mono;

@Component
@RequiredArgsConstructor
public class UserFacade {

    private final UserService userService;

    public Mono<UserResult> createUser(UserCriteria criteria) {
        UserCommand userCommand = new UserCommand(criteria);
        Mono<UserInfo> userInfo = userService.save(userCommand);

        return userInfo.map(UserResult::new);
    }
}

File: warmup-example/src/main/kotlin/com/example/warmupexample/application/facade/user/UserFacade.kt
package com.example.warmupexample.application.facade.user

import com.example.warmupexample.domain.user.service.UserService
import com.example.warmupexample.domain.user.service.toResult
import org.springframework.stereotype.Component

@Component
class UserFacade(
    private val userService: UserService,
) {
    fun getUsers(): List<UserResult> {
        val users = userService.getUsers()
        return users.map { it.toResult() }
    }
}